crates/store/src/blocks.rs

```diff
@@ -0,0 +1,45 @@
+use std::{future::Future, path::PathBuf};
+
+pub trait BlockStorage: Send + Sync + 'static {
+    fn save_block(
+        &self,
+        block_num: u32,
+        data: &[u8],
+    ) -> impl Future<Output = Result<(), std::io::Error>> + Send;
+
+    fn load_block(
+        &self,
+        block_num: u32,
+    ) -> impl Future<Output = Result<Option<Vec<u8>>, std::io::Error>> + Send;
+}
+
+#[derive(Debug)]
+pub struct BlockStorageDefault {
+    blocks_dir: PathBuf,
+}
+
+impl BlockStorageDefault {
+    pub async fn new(blocks_dir: PathBuf) -> Result<Self, std::io::Error> {
+        tokio::fs::create_dir_all(&blocks_dir).await?;
+
+        Ok(Self { blocks_dir })
+    }
+
+    fn block_path(&self, block_num: u32) -> PathBuf {
+        self.blocks_dir.join(format!("{block_num:08x}.bin"))
```

Comment:
Couple nits:

- I would change the extension from `.bin` to `.dat` because `.bin` could be a bit ambiguous.
- I would just use `block_` prefix (instead of `block_num:`). So, `block_12345.dat`.